src/Entities/Entity.cpp

```diff
@@ -1576,47 +1577,70 @@ void cEntity::DoMoveToWorld(const sWorldChangeInfo & a_WorldChangeInfo)
 
 
 
-bool cEntity::MoveToWorld(cWorld * a_World, Vector3d a_NewPosition, bool a_SetPortalCooldown, bool a_ShouldSendRespawn)
+bool cEntity::MoveToWorld(cWorld & a_World, Vector3d a_NewPosition, bool a_SetPortalCooldown, bool a_ShouldSendRespawn)
 {
-	ASSERT(a_World != nullptr);
-
 	// Ask the plugins if the entity is allowed to change world
```

Comment:
```suggestion
	ASSERT(m_World.GetChunkMap().GetCS().IsLockedByCurrentThread());
	
	// Ask the plugins if the entity is allowed to change world
```
Try running the `/portal` command with this.

Reply:
It is not locked.

I'm not convinced it has to be though: apart from the thread-safe QueueTask, MoveToWorld only accesses entity variables, and entities can only be accessed from within the tick thread.

Reply:
>entities can only be accessed from within the tick thread.

That's simply not true. `cRoot:GetWorld` means any plugin can access the entities, no matter what thread they are on.

Reply:
Though this is incorrect behaviour if the plugin does so, just like if they store an entity reference across ticks and crash the server when it goes stale. I'm certain there are multiple places where we assume world state is only updated during a tick, and anything that violates this needs to be fixed. I suppose what I'm getting at is that there's no point in making MoveToWorld safe if every other cEntity-related function isn't.

Reply:
>there's no point in making MoveToWorld safe if every other cEntity-related function isn't.

True, the main issue with `MoveToWorld` should be fixed by #4647